### PR TITLE
添加数据校验

### PR DESCRIPTION
**Pull request by KOBEliever**

### CODE REVIEW

File: Service/defect/src/main/java/com/cityu/defect/controller/DefectInfoController.java [hunk 1]
```diff
@@ -58,6 +58,10 @@ public BaseResponse<List<DefectInfoVO>> searchDefectInfoVO(@RequestBody DefectIn
         if(loginUser == null){
             throw new BusinessException(ErrorCode.NOT_LOGIN_ERROR);
         }
+        //需要传递userId
+        if(defectInfoQueryRequest.getUserId()==null){
+            throw new BusinessException(40001,"Your request have to contain your userId");
```
Comment (wychen-null) [inline Service/defect/src/main/java/com/cityu/defect/controller/DefectInfoController.java:63]: Do not use constant numbers in your code. Write 40001 into ErrorCode file

File: Service/defect/src/main/java/com/cityu/defect/service/impl/DefectInfoServiceImpl.java [hunk 2]
```diff
@@ -47,6 +47,13 @@ public List<DefectInfo> getQueryWrapper(DefectInfoQueryRequest defectInfoQueryRe
         if (defectInfoQueryRequest == null) {
             throw new BusinessException(ErrorCode.PARAMS_ERROR, "请求参数不能为空");
         }
+        //TODO 数据校验
```
Comment (wychen-null) [inline Service/defect/src/main/java/com/cityu/defect/service/impl/DefectInfoServiceImpl.java:50]: delete TODO

**Review by wychen-null — COMMENTED** [review]
modify these issues

File: Service/defect/src/main/java/com/cityu/defect/controller/DefectInfoController.java (hunk 1 above)
Comment (KOBEliever) [reply Service/defect/src/main/java/com/cityu/defect/controller/DefectInfoController.java:63]: ok, my bad

File: Service/defect/src/main/java/com/cityu/defect/service/impl/DefectInfoServiceImpl.java (hunk 2 above)
Comment (KOBEliever) [reply Service/defect/src/main/java/com/cityu/defect/service/impl/DefectInfoServiceImpl.java:50]: ok

**Comment by KOBEliever** [comment]
code review finished

**Comment by wychen-null** [comment]
perfect

**Pull request merged**